src/rust/ensogl/lib/select/src/component.rs

```diff
@@ -0,0 +1,403 @@
+                let y_range = (-size.y / 2.0)..=(size.y / 2.0);
+                x_range.contains(&pos_obj_space.x) && y_range.contains(&pos_obj_space.y)
+            }));
+            mouse_moved       <- mouse.distance.map(|dist| *dist > std::f32::EPSILON);
+            mouse_y_in_scroll <- mouse.position.map(f!([model,scene](pos) {
+                scene.screen_to_object_space(&model.scrolled_area,*pos).y
+            }));
+            mouse_pointed_entry <- mouse_y_in_scroll.map(f!((y)
+                model.entries.entry_at_y_position(*y).entry())
+            );
+
+
+            // === Selected Entry ===
+            frp.source.selected_entry <+ frp.select_entry.map(|id| Some(*id));
+
+            selection_jump_on_one_up  <- frp.move_selection_up.constant(-1);
+            selection_jump_on_page_up <- frp.move_selection_page_up.map(f!([model](())
+                -(model.entries.visible_entry_count() as isize))
+            );
+            selection_jump_on_one_down <- frp.move_selection_down.constant(1);
+            selection_jump_on_page_down <- frp.move_selection_page_down.map(f!([model](())
+                model.entries.visible_entry_count() as isize)
+            );
+            selection_jump_up   <- any(selection_jump_on_one_up,selection_jump_on_page_up);
+            selection_jump_down <- any(selection_jump_on_one_down,selection_jump_on_page_down);
+            selected_entry_after_jump_up <- selection_jump_up.map2(&frp.selected_entry,
+                |jump,id| {
+                    let id = id.as_ref()?;
+                    if *id == 0 && *jump != 0 { None                                    }
+                    else                      { Some(id.saturating_sub(-jump as usize)) }
+                }
+            );
+            selected_entry_after_moving_first <- frp.move_selection_to_first.map(f!([model](())
+                (model.entries.entry_count() > 0).and_option(Some(0))
+            ));
+            selected_entry_after_moving_last  <- frp.move_selection_to_last.map(f!([model] (())
+                model.entries.entry_count().checked_sub(1)
+            ));
+            selected_entry_after_jump_down <- selection_jump_down.map2(&frp.selected_entry,
+                f!([model](jump,id) {
+                    let max_entry = model.entries.entry_count().checked_sub(1)?;
+                    Some(id.map_or(0, |id| id+(*jump as usize)).min(max_entry))
+                }
+            ));
+            selected_entry_after_move_up <- any(selected_entry_after_jump_up,
+                selected_entry_after_moving_first);
+            selected_entry_after_move_down <- any(selected_entry_after_jump_down,
+                selected_entry_after_moving_last);
+            selected_entry_after_move <- any(&selected_entry_after_move_up,
+                &selected_entry_after_move_down);
+            mouse_selected_entry <- mouse_pointed_entry.gate(&mouse_in).gate(&mouse_moved);
```

Comment:
imo it owuld be much more clean if you break the line directly after `<-` instead, moving `any(...)` as a whole expression to new line.